app/components/public/ticket-list.js

```diff
@@ -94,11 +106,20 @@ export default Component.extend(FormMixin, {
           let tickets = await discountCode.get('tickets');
           tickets.forEach(ticket => {
             let ticketPrice = ticket.get('price');
+            let taxRate = ticket.get('event.tax.rate');
             if (discountType === 'amount') {
               ticket.set('discount', Math.min(ticketPrice, discountValue));
+              if (taxRate) {
+                let ticketPriceWithTax = (ticketPrice - ticket.discount) * (1 + taxRate / 100);
+                ticket.set('ticketPriceWithTax', ticketPriceWithTax);
+              }
               this.discountedTickets.addObject(ticket);
             } else {
               ticket.set('discount', ticketPrice * (discountValue / 100));
+              if (taxRate) {
```

Comment:
This code segment is repetitive. Change it to this:
```js
let discount = discountType === 'amount' ? Math.min(ticketPrice, discountValue) : ticketPrice * (discountValue / 100);
ticket.set('discount', discount);
if (taxRate) {
  let ticketPriceWithTax = (ticketPrice - ticket.discount) * (1 + taxRate / 100);
  ticket.set('ticketPriceWithTax', ticketPriceWithTax);
}
this.discountedTickets.addObject(ticket);
```

Reply:
Merge this code segment with the upper one (which it repeats)

Reply:
@pradeepgangwar Sure! Will do that in next few minutes :smile: